nearswap/src/twap.rs

```diff
@@ -0,0 +1,553 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::{U128, U64};
+use near_sdk::collections::{Vector};
+use near_sdk::{env};
+use std::convert::{TryFrom,TryInto};
+
+use std::fmt;
+use crate::constants::*;
+use crate::pool::*;
+use crate::*;
+
+#[derive(Debug)]
+pub enum Mean {
+    M_1MIN,
+    M_5MIN,
+    M_1H,
+    M_12H,
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct Twap {
+    // To check if array is populated or not
+    // note: it can be used later when we will have variable length storage array/
+    populated: usize,
```

Comment:
please rename and update the comment. `populated` indicates a boolean type. This is not boolean type.

Reply:
Maybe we don't need this variable? We can move `initialize()` code to `new()`

Reply:
we may need this, because new gives blank vector of observations, and initialize() store first value, when pool is initialized swap is not performed therefore we cannot have it in new().
when the swap is performed we will call log_observation function which will handle `initialize()` and `write()`

Reply:
removed this variable, added a check in `log_observation()` if observation array is empty then `intiialize()` else `write()`